### PR TITLE
Enables database migration generation past the initial

### PR DESCRIPTION
**Pull request by joeconwaystk**
Addresses #127.

Requires the project to depend on version >2.0.3.

Also cleans up testing around CLI tools in general. Sorry it is way too many changes for one PR

### CODE REVIEW

**Comment by anachlas** [comment]
I am so excited for the functionality in this PR!

File: lib/src/commands/db_generate.dart [hunk 1]
```diff
@@ -12,53 +12,63 @@ class CLIDatabaseGenerate extends CLICommand
...
+      var inputSchema = new Schema.fromMap(values["initialSchema"]);
       var dataModel = new ManagedDataModel.fromCurrentMirrorSystem();
       var schema = new Schema.fromDataModel(dataModel);
+      var changeList = <String>[];
+
+      var source;
+      if (currentMirrorSystem().libraries.containsKey(Uri.parse("package:aqueduct/src/db/schema/migration_builder.dart"))) {
+        // This runs on 2.0.3 and later. Previous project versions will fallback to outdated version.
+        source = MigrationBuilder.sourceForSchemaUpgrade(
+            inputSchema, schema, 1, changeList: changeList);
```
Comment (anachlas) [inline lib/src/commands/db_generate.dart]: Hardcoded migration version number

Comment (joeconwaystk) [inline lib/src/commands/db_generate.dart]: ooo thanks

File: lib/src/commands/db_generate.dart [hunk 2]
```diff
@@ -12,53 +12,64 @@ class CLIDatabaseGenerate extends CLICommand
...
-      return 0;
+      versionNumber  = versionNumberFromFile(files.last) + 1;
+      newMigrationFile = new File.fromUri(migrationDirectory.uri.resolve(
+          "${"$versionNumber".padLeft(8, "0")}_Unnamed.migration.dart"));
     }
 
-    var file = new File.fromUri(
-        migrationDirectory.uri.resolve("00000001_Initial.migration.dart"));
-    var source = await generateMigrationSource();
+    var source = await generateMigrationSource(await schemaMapFromExistingMigrationFiles());
```
Comment (anachlas) [inline lib/src/commands/db_generate.dart]: Need to pass version number here

**Comment by erikist** [comment]
How exciting

**Comment by joeconwaystk** [comment]
yeah, got a failing test somewhere else that'll take me a bit to fix

**Pull request merged**